tests/functional_test_base.py

```diff
@@ -81,22 +80,37 @@ def configure_moto(self):
         self.boto_connection.create_bucket("test")
         self.boto_connection.create_bucket("bucket2")
         self.test_bucket = self.boto_connection.get_bucket("test")
-        self.configure_artifacts()
+        self.setup_artifacts()
         self.create_auth_key()
 
-    def configure_artifacts(self):
+    def setup_artifacts(self):
         key = Key(self.test_bucket, "test")
         key.set_contents_from_string("hello world")
         nested_key = Key(self.test_bucket, "/dir/dir2/dir3/nest-test")
         nested_key.set_contents_from_string("hello world")
-        # Metadata for artifacts
-        meta_key = Key(self.test_bucket, "_metadata_test.yaml")
-        meta_key.set_contents_from_string(yaml.dump(meta_utils.get_meta()))
-        nest_meta_key = Key(self.test_bucket, "/dir/dir2/dir3/_metadata_nest-test.yaml")
-        nest_meta_key.set_contents_from_string("")
         artifact_list = Key(self.test_bucket, "/dir/dir2/dir3/dir4/test5")
         artifact_list.set_contents_from_string("")
 
+    def setup_metadata(self):
+        self.add_metadata("test", "/", "test")
+        self.add_metadata("test", "/dir/dir2/dir3/", "nest-test")
+
+    def add_metadata(self, bucket_name, artifact_path, artifact_name, metadata=None):
```

Comment:
Could you just use the full path and then ResourceIdentity here as well?  We already do this, so the damage is done.  May as well take advantage of it